packages/main/src/StepInput.ts

```diff
@@ -441,12 +449,19 @@ class StepInput extends UI5Element implements IFormElement {
 	}
 
 	_updateValueState() {
-		const valid = !((this.min !== undefined && this.value < this.min) || (this.max !== undefined && this.value > this.max));
+		const isWithinRange = (this.min === undefined || Number(this.input.value) >= this.min)
```

Comment:
Because we need to check the value that the user has provided at the moment, while the `this.value` is getting modified and the check would not be relevant at the end of the interaction.